### PR TITLE
feat: add regex lab

### LINKED ISSUE #458: chore: add lab for regex in frontend cert
We need a lab for them to do on Regular expressions.
Here is the workshop they will do before this
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-regular-expressions-by-building-a-spam-filter/step-35

They will also know all of the fundamentals (strings, functions, objects, arrays, etc)
They will also know about higher order functions like map, filter, reduce.
And they will know DOM basics like getElementById, querySelector, querySelectorAll, click events, change events. 

## Acceptance criteria

- [ ] create a new folder of the project in the `frontend-cert/js-projects` directory
- [ ] add your completed project in an `script.js` file in that new folder
- [ ] create a markdown file for your proposed user stories. You should call it `user-stories.md` and place it in your new project folder

### PR DESCRIPTION
**Pull request by Dario-DC**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #458

### CODE REVIEW

File: frontend-cert/js-projects/markdown-parser/user-stories.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+1. You should have a function named `convertMarkdown` that takes no parameter.
+1. The `convertMarkdown` function should convert the makdown input from `#markdown-input` into HTML and should return a string containing the HTML code.
```
Comment (zairahira) [inline frontend-cert/js-projects/markdown-parser/user-stories.md:2]:
```suggestion
1. The `convertMarkdown` function should convert the markdown input from `#markdown-input` into HTML and should return a string containing the HTML code.
```

**Review by zairahira — COMMENTED** [review]
I was going through the prototype. If I understood this right, the result should be raw HTML right?

For example `### this is heading 3` should be `<h3>this is heading 3</h3>`?

When I ran the code, I got the rendered HTML as the output:

image

**Comment by Dario-DC** [comment]
@zairahira my idea was to have the HTML rendered on the page (see the last user story). I guess we can change the code if we don't want that.

**Comment by zairahira** [comment]
I'd suggest it is better to show the raw HTML because visually, there seems to be no difference in rendering either markdown or HTML.

**Comment by jdwilkin4** [comment]
Ultimately, I think it should be up to you if you want to update the functionality to convert markdown to raw HTML like zaira suggested or update the language to better convey a markdown parser

**Comment by Dario-DC** [comment]
What about something like this?
image

And with this change I would keep the table.

**Comment by jdwilkin4** [comment]
Yeah, that works for me

**Comment by zairahira** [comment]
This is looking good @Dario-DC.

**Comment by Dario-DC** [comment]
Prototype updated. Let me know if you think I can go ahead and merge this PR.

**Review by jdwilkin4 — APPROVED** [review]
looks good, let's get this merged in.

Are you are going to work on this lab for the main repo? Or do I need to update the priority list of items so someone else can pick this up?

**Comment by Dario-DC** [comment]
> looks good, let's get this merged in.
> 
> Are you are going to work on this lab for the main repo? Or do I need to update the priority list of items so someone else can pick this up?

I'll start to work on it tomorrow :+1:

**Pull request merged**